### ISSUE TITLE
Java runner isn't compatible with Bazel > 0.21

### ISSUE THREAD

**Issue by achew22** [bazelbuild/bazel-integration-testing#106]
Sample run output: https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/57e26e8a-4dfe-4978-83eb-717a41f91849/javatests/build/bazel/tests/integration/BazelBaseTestCaseTest/bazel0.21.0/attempt_1.log

```
STDERR:
    $TEST_TMPDIR defined: output root default is '/home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb' and max_idle_secs default is '15'.
    Starting local Bazel server and connecting to it...
    INFO: Invocation ID: 417d9c78-6445-403c-8056-d4b8deee9464
    Loading: 
    Loading: 0 packages loaded
    Loading: 0 packages loaded
    Analyzing: 13 targets (1 packages loaded)
    Analyzing: 13 targets (1 packages loaded, 0 targets configured)
    Analyzing: 13 targets (3 packages loaded, 16 targets configured)
    Analyzing: 13 targets (18 packages loaded, 353 targets configured)
    Analyzing: 13 targets (18 packages loaded, 353 targets configured)
    ERROR: /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/0e3c39548b556d9f0eec702c5dd0dd08/external/bazel_tools/tools/jdk/BUILD:305:1: no such package '@remotejdk_linux//': com.google.devtools.build.lib.vfs.FileAccessException: /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/cache/content_addressable/sha256/f27cb933de4f9e7fe9a703486cf44c84bc8e9f138be0c270c9e5716a32367e87 (Permission denied) and referenced by '@bazel_tools//tools/jdk:remote_jdk'
    ERROR: Analysis of target '//:IntegrationTestSuiteTest/bazel0.19.1' failed; build aborted: Analysis failed
    INFO: Elapsed time: 9.948s
    INFO: 0 processes.
    FAILED: Build did NOT complete successfully (18 packages loaded, 353 targets configured)
    ERROR: Couldn't start the build. Unable to run tests
    FAILED: Build did NOT complete successfully (18 packages loaded, 353 targets configured)
WORKSPACE CONTENTS:
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build/bazel
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build/bazel/tests
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build/bazel/tests/integration
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build/bazel/tests/integration/libworkspace_driver.jar
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/java/build/bazel/tests/integration/BUILD.bazel
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/IntegrationTestSuiteTest.java
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/WORKSPACE
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/BUILD
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/go
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/go/bazel_integration_test.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/go/BUILD.bazel
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/BUILD
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/repositories.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/bazel_py_integration_test.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/import.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/common.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/bazel_hash_dict.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/bazel_java_integration_test.bzl
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/tools/create_bazel_hashdict.py
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace4291130982696570630/bazel_integration_test.bzl
==> non-zero exit code
	at build.bazel.tests.integration.BazelCommand$Builder.mustRunSuccessfully(BazelCommand.java:172)
	at build.bazel.tests.integration.BazelBaseTestCaseTest.testTestSuiteExists(BazelBaseTestCaseTest.java:57)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at com.google.testing.junit.runner.internal.junit4.CancellableRequestFactory$CancellableRunner.run(CancellableRequestFactory.java:89)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:112)
	at com.google.testing.junit.runner.BazelTestRunner.runTestsInSuite(BazelTestRunner.java:144)
	at com.google.testing.junit.runner.BazelTestRunner.main(BazelTestRunner.java:82)
2) testRunningBazelInGivenRelativeDirectory(build.bazel.tests.integration.BazelBaseTestCaseTest)
java.lang.RuntimeException: BAZEL COMMAND: [run, bar]
EXIT CODE: 1
STDOUT:
    
STDERR:
    $TEST_TMPDIR defined: output root default is '/home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb' and max_idle_secs default is '15'.
    Starting local Bazel server and connecting to it...
    INFO: Invocation ID: 693ab57d-0a6a-47fe-9f03-fd59c3b9641e
    Loading: 
    Loading: 0 packages loaded
    Loading: 0 packages loaded
    Analyzing: target //foo:bar (1 packages loaded, 0 targets configured)
    Analyzing: target //foo:bar (6 packages loaded, 22 targets configured)
    Analyzing: target //foo:bar (8 packages loaded, 45 targets configured)
    Analyzing: target //foo:bar (16 packages loaded, 120 targets configured)
    Analyzing: target //foo:bar (17 packages loaded, 326 targets configured)
    Analyzing: target //foo:bar (17 packages loaded, 326 targets configured)
    ERROR: /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/bff644596ec7805088952b390a4e3bd8/external/bazel_tools/tools/jdk/BUILD:305:1: no such package '@remotejdk_linux//': com.google.devtools.build.lib.vfs.FileAccessException: /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/cache/content_addressable/sha256/f27cb933de4f9e7fe9a703486cf44c84bc8e9f138be0c270c9e5716a32367e87 (Permission denied) and referenced by '@bazel_tools//tools/jdk:remote_jdk'
    ERROR: Analysis of target '//foo:bar' failed; build aborted: Analysis failed
    INFO: Elapsed time: 11.836s
    INFO: 0 processes.
    FAILED: Build did NOT complete successfully (17 packages loaded, 326 targets configured)
    ERROR: Build failed. Not running target
    FAILED: Build did NOT complete successfully (17 packages loaded, 326 targets configured)
WORKSPACE CONTENTS:
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace1673767554258561207
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace1673767554258561207/WORKSPACE
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace1673767554258561207/foo
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace1673767554258561207/foo/bar.sh
    /home/bazel/.cache/bazel/_bazel_bazel/ec321eb2cc2d0f8f91b676b6d4c66c29/sandbox/linux-sandbox/57/execroot/build_bazel_integration_testing/_tmp/5779ebb4e6ed1d2af99650167aeb7cfb/workspace1673767554258561207/foo/BUILD
==> non-zero exit code
	at build.bazel.tests.integration.BazelCommand$Builder.mustRunSuccessfully(BazelCommand.java:172)
	at build.bazel.tests.integration.BazelBaseTestCaseTest.testRunningBazelInGivenRelativeDirectory(BazelBaseTestCaseTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at com.google.testing.junit.runner.internal.junit4.CancellableRequestFactory$CancellableRunner.run(CancellableRequestFactory.java:89)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:112)
	at com.google.testing.junit.runner.BazelTestRunner.runTestsInSuite(BazelTestRunner.java:144)
	at com.google.testing.junit.runner.BazelTestRunner.main(BazelTestRunner.java:82)

FAILURES!!!
Tests run: 4,  Failures: 2
```

**Comment by ittaiz** [comment]
I’m on it

**Comment by ittaiz** [comment]
fixed by #120
